pkg/package-server/server/server.go

```diff
@@ -184,6 +185,9 @@ func (o *PackageServerOptions) Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	if config == nil {
+		return fmt.Errorf("failed to construct an API server config")
+	}
```

Comment:
Adding more context to the PR/commit message:

- When attempting to initialize the delegating authentication options in https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/package-server/server/server.go#L130-L142 and no cluster is present, and SIGINT is sent to that package-server process, a timeout error will be returned during that poll operation as that signal is caught.
- The lastApplyErr is returned in https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/package-server/server/server.go#L140 despite that error being nil
- Line 184 initializes a `config, err := o.Config(ctx) ` variables
- That error will be evaluated as null as the lastApplyErr is always being returned and an empty config object is returned
- Line 191 attempts to access a nested field in an empty object, which produces a panic

Likely can also handle this in the o.Config method and instead double-check we're returning the correct error, so that call sites of this method can be ensured that the config structure is populated when an error is not nil.

Reply:
I was about to suggest the `o.Config` solution. Seems to be the right place to put it and the right contract to give the developers for this method.